state/transaction.go

```diff
@@ -1054,31 +1038,54 @@ func (s *State) EstimateGas(transaction *types.Transaction, senderAddress common
 			}
 		}
 
-		availableBalance = new(big.Int).Set(senderBalance)
-
+		availableBalance := new(big.Int).Set(senderBalance)
+		// check if the account balance is at least bigger than the tx Value
```

Comment:
available balance to send `value` should be check for `defaultAdress` & if the `gasPrice` is not set as well